src/main/java/uk/gov/companieshouse/efs/api/email/mapper/EmailMapperFactory.java

```diff
@@ -255,31 +219,14 @@ public Builder withPaymentReportEmailMapper(final PaymentReportEmailMapper payme
          * @return a {@code EmailMapperFactory} built with parameters of this {@code EmailMapperFactory.Builder}
          */
         public EmailMapperFactory build() {
-            checkNotNull(this.acceptEmailMapper, "'acceptEmailMapper' must not be null");
-            checkNotNull(this.confirmationEmailMapper,
-                "'confirmationEmailMapper' must not be null");
-            checkNotNull(this.paymentFailedEmailMapper,
-                "'paymentFailedEmailMapper' must not be null");
-            checkNotNull(this.delayedSubmissionBusinessEmailMapper,
-                "'delayedSubmissionBusinessEmailMapper' must not be null");
-            checkNotNull(this.delayedSubmissionSupportEmailMapper, "'delayedSubmissionSupportEmailMapper' must not be null");
-            checkNotNull(this.delayedSH19SameDaySubmissionSupportEmailMapper,
-                "'this.delayedSH19SameDaySubmissionSupportEmailMapper' must not be null");
-            checkNotNull(this.internalAvFailedEmailMapper, "'internalAVFailedEmailMapper' must not be null");
-            checkNotNull(this.internalFailedConversionEmailMapper, "'internalFailedConversionEmailMapper' must not be null");
-            checkNotNull(this.internalSubmissionEmailMapper, "'internalSubmissionEmailMapper' must not be null");
-            checkNotNull(this.paymentReportEmailMapper, "'paymentReportEmailMapper' must not be null");
-            checkNotNull(this.rejectEmailMapper, "'rejectEmailMapper' must not be null");
-
-            return new EmailMapperFactory(this);
-        }
+            EmailMapperFactory data = new EmailMapperFactory();
+
+            buildSteps.forEach(step -> step.accept(data));
```

Comment:
The same required field validation checks `checkNotNull(...` need to be done here, not removed.
Best to move into separate method:
```
validate(data);
```
then
```
private void validate(EmailMapperFactory data) {
  // checkNotNull() for all required fields...
}
```